python_modules/dagster/dagster/core/types/config.py

```diff
@@ -238,7 +243,8 @@ def __init__(self, config_value, python_value=None, description=None):
 
 class ConfigEnum(ConfigType):
     def __init__(self, name, enum_values):
-        super(ConfigEnum, self).__init__(name=check.str_param(name, 'name'))
+        check.str_param(name, 'name')
+        super(ConfigEnum, self).__init__(key=name, name=name)
```

Comment:
This is a bit of a sidenote but it'd be nice if the key was never the same as the `name` just so that API consumers know they can't rely on them to be equivalent. (You might glance over a particular GraphQL `types` query and assume they're the same if you don't get any anonymous types). We could prefix them with `k_` or something like that and still use the name to avoid needing a hashing function.